src/analysis/analysis-service.js

```diff
@@ -80,10 +80,6 @@ AnalysisService.prototype._onAnalysisRemoved = function (analysis) {
   analysis.unbind('destroy', this._onAnalysisRemoved);
 };
 
-AnalysisService.prototype.findNodeById = function (id) {
```

Comment:
Why not just change the signature of this method and use `AnalysisService.getUniqueAnalysesNodes` internally? I'm a bit concerned about defining this method to the instance of the `AnalysisService` at runtime [here](https://github.com/CartoDB/cartodb.js/pull/1815/files#diff-f99ed960498d5ea8717b3e74d54bfb01R147).

Reply:
this is a temporal fix, all this logic will be moved to the engine.

The engine will have both (layers and dataviews list) allowing the rest of our code to be state-less.

So the engine will call `AnalysisService.findByID(id, layersCollection, dataviewsCollection)``
in the meantime we expose this functionality through visModel to allow backward compatibility